gmail/src/models/state.ts

```diff
@@ -0,0 +1,196 @@
+import { Email } from "./email";
+import { Partner } from "./partner";
+import { Lead } from "./lead";
+import { ErrorMessage } from "./error_message";
+import { getAccessToken, getServiceUrl } from "../services/odoo_auth";
+
+/**
+ * Object which contains all data for the application.
+ *
+ * In App-Script, all event handler are function and not method. We can only pass string
+ * as arguments. So this object is serialized, then given to the event handler and then
+ * unserialize to retrieve the original object.
+ *
+ * That's how we manage the state of the application without performing a big amount of
+ * read / write in the cache.
+ */
+
+export class State {
+    partner: Partner;
+    email: Email;
+    partners: Partner[];
+    error: ErrorMessage;
```

Comment:
While your model classes are pretty self-explanatory, this one is more cryptic.

What is the "partner" ? The current partner of the opened emails?
What are "partners" ?
...

Could be nice to add some explanations for every state variable.